win32/src/socket_transport_win32.c

```diff
@@ -0,0 +1,720 @@
+
+                u_long iMode = 1;
+                if (ioctlsocket(client_socket, FIONBIO, &iMode) != 0)
+                {
+                    LogLastError("Failure: ioctlsocket failure.");
+                    result = INVALID_SOCKET;
+                }
+                // Codes_SOCKET_TRANSPORT_WIN32_09_016: [ socket_transport_connect shall call connect_to_endpoint with the connection_timeout_ms to connect to the endpoint. ]
+                else if (connect_to_endpoint(client_socket, addrInfo, connection_timeout) != 0)
+                {
+                    LogLastError("Connection failure. Server: %s:%" PRIu16 " Machine Name: %s.", hostname, port, MU_P_OR_NULL(addrInfo->ai_canonname));
+                    result = INVALID_SOCKET;
+                }
+                else
+                {
+                    freeaddrinfo(addrInfo);
+                    result = client_socket;
+                    goto all_ok;
+                }
+                freeaddrinfo(addrInfo);
+            }
+        }
+        closesocket(client_socket);
+    }
+all_ok:
+    return result;
+}
+
+SOCKET_TRANSPORT_HANDLE socket_transport_create(SOCKET_TYPE type)
+{
+    SOCKET_TRANSPORT* result;
+
+    // Codes_SOCKET_TRANSPORT_WIN32_09_001: [ socket_transport_create shall ensure type is either SOCKET_CLIENT, or SOCKET_SERVER. ]
+    if (type != SOCKET_CLIENT && type != SOCKET_SERVER)
+    {
+        LogError("Invalid socket type specified: %" PRI_MU_ENUM "", MU_ENUM_VALUE(SOCKET_TYPE, type));
+        result = NULL;
+    }
+
+    else
+    {
+        // Codes_SOCKET_TRANSPORT_WIN32_09_002: [ socket_transport_create shall allocate a new SOCKET_TRANSPORT object. ]
+        result = malloc(sizeof(SOCKET_TRANSPORT));
+        if (result == NULL)
+        {
+            LogError("failure allocating SOCKET_TRANSPORT: %zu", sizeof(SOCKET_TRANSPORT));
+        }
+        else
+        {
+            // Codes_SOCKET_TRANSPORT_WIN32_09_003: [ socket_transport_create shall call sm_create to create a sm object. ]
+            result->sm = sm_create("Socket_transport_win32");
+            if (result->sm == NULL)
+            {
+                LogError("sm_create failed.");
+            }
+            else
+            {
+                result->type = type;
+                goto all_ok;
+            }
+            free(result);
+            // Codes_SOCKET_TRANSPORT_WIN32_09_004: [ On any failure socket_transport_create shall return NULL. ]
+            result = NULL;
+        }
+    }
+all_ok:
+    // Codes_SOCKET_TRANSPORT_WIN32_09_005: [ On success socket_transport_create shall return SOCKET_TRANSPORT_HANDLE. ]
+    return result;
+}
+
+void socket_transport_destroy(SOCKET_TRANSPORT_HANDLE socket_transport)
+{
+    // Codes_SOCKET_TRANSPORT_WIN32_09_006: [ If socket_transport is NULL socket_transport_destroy shall return. ]
+    if (socket_transport == NULL)
+    {
+        // Do nothing
+        return;
```

Comment:
>return; [](http://example.com/codeflow?start=8&length=7)

Don't return in the middle of the function, return should only be at the last line of a function.  In this situation, just let it return naturally at the end, no return needed.